apps/angular/src/core/interceptors/authentication.interceptor.ts

```diff
@@ -0,0 +1,61 @@
+import { HttpEvent, HttpInterceptor, HttpHandler, HttpRequest, HttpStatusCode } from '@angular/common/http';
+import { Injectable, inject } from '@angular/core';
+import { Observable, map, switchMap } from 'rxjs';
+
+import { AuthService } from '../services/auth.service';
+import { catchApiError } from '../utils/rxjs/catch-api-error';
+import { TokensStorageService } from '../services/tokens-storage.service';
+import { ApiUriBuilder } from '../services/api-uri-builder';
+
+/** Interceptor for request authentication. */
+@Injectable()
+export class AuthenticationInterceptor implements HttpInterceptor {
+
+	private readonly authService = inject(AuthService);
+
+	private readonly tokensStorageService = inject(TokensStorageService);
+
+	private readonly apiUriBuilder = inject(ApiUriBuilder);
+
+	/** @inheritdoc */
+	public intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
+		if (this.isAuthenticationRequest(request)) {
+			return next.handle(request);
+		}
+
+		const authenticatedRequest = this.addAuthentication(request);
+
+		return next.handle(authenticatedRequest).pipe(
+			catchApiError((apiError, throwApiError$) => apiError.statusCode !== HttpStatusCode.Unauthorized ?
+				throwApiError$ :
+				this.refreshToken(request).pipe(
+					switchMap(refreshedRequest => next.handle(refreshedRequest)),
+				)),
+		);
+	}
+
+	private isAuthenticationRequest(request: HttpRequest<unknown>): boolean {
+		return request.url === this.apiUriBuilder.buildLoginUri() ||
+			request.url === this.apiUriBuilder.buildRegistrationUri() ||
+			request.url === this.apiUriBuilder.buildRefreshUri();
```

Comment:
Let's move this logic to 'ApiUriBuilder'. My suggestion will be to make something like this:
```suggestion
		return this.apiUriBuilder.getAuthUrls().includes(request.url);
```

Alternatively you can use `startsWith` utility if all Auth urls have the same prefix:
```suggestion
		return request.url.startsWith(this.apiUriBuilder.getAuthBaseUri());
```